scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,74 @@ def visit_functiondef(self, node):
     # this line makes it work for async functions
     visit_asyncfunctiondef = visit_functiondef
 
+class CheckApiVersion(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    name = "check-api-version"
+    priority = -1
+    msgs = {
+        "C4748": (
+            "Check api version",
+            "api-version-not-keyword",
+            "Acceot a keyword argument called api_version.",
+        ),
+    }
+    options = (
+        (
+            "ignore-api-version-incorrect",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow for no keyword api version.",
+            },
+        ),
+    )
+    ignore_clients = ["PipelineClient", "AsyncPipelineClient", "ARMPipelineClient", "AsyncARMPipelineClient"]
+
+    def __init__(self, linter=None):
+        super(CheckApiVersion, self).__init__(linter)             
+
+    def visit_classdef(self, node):
+        """Visits every class in file and checks if it is a client.
+        If it is a client, it checks that there is an api_version keyword.
+
+        :param node: class node
+        :type node: ast.ClassDef
+        :return: None
+        """
+
+        #Need to take into account python 2 versus version 3 of python
+        # Do accept an optional api_version kwargs
+
+        try:
+            api_version = False
+            
+            if node.name.endswith("Client") and node.name not in self.ignore_clients:
+                for func in node.body:
+                    if isinstance(func,astroid.FunctionDef):
+                        if func.name == '__init__':
+                            if node.doc:
+                                kwargs_found = node.doc.split("keyword")
+                                if "api_version" in " ".join(kwargs_found[1::]):
+                                    api_version = True
```

Comment:
should this segment be moved out of the isinstance block since we're just looking at the ast.ClassDef's docstring?